### PR TITLE
Adds a base class for key/value containers that can be saved and restored

### PR DESCRIPTION
**Pull request by happz**
This functionality is used throughout the code, adding a class others
could use to gain access to this functionality.

Right now, targetting guests and their load/save workflow, but in the
future, other objects like plugins would get converted too.

### CODE REVIEW

**Comment by happz** [comment]
https://github.com/teemtee/tmt/pull/1247/files depends on this PR, and converts `Guest` classes and data blobs used in their creation.

**Review by thrix — APPROVED** [review]
LGTM

**Review by psss — COMMENTED** [review]
Looks very good. Just a question around generator methods naming.

File: tmt/utils.py [hunk 1]
```diff
@@ -1127,6 +1128,147 @@ def yaml_to_dict(data: Any,
...
+
+    @classmethod
+    def iter_key_names(cls) -> Generator[str, None, None]:
+        """ Iterate over key names """
+
+        for field in dataclasses.fields(cls):
+            yield field.name
+
+    @classmethod
+    def iter_keys(cls) -> Generator[Tuple[str, Any], None, None]:
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/utils.py]: Nevermind, added `values()` too, in [cfb5922](https://github.com/teemtee/tmt/pull/1246/commits/cfb5922b66f9bddd3a131730e339b13ce7938f56).

File: tmt/utils.py [hunk 2]
```diff
@@ -1127,6 +1128,147 @@ def yaml_to_dict(data: Any,
...
+    'SerializableContainerDerivedType',
+    bound='SerializableContainer')
+
+
+@dataclasses.dataclass
+class SerializableContainer:
+    """ A mixin class for objects that may be saved in files and restored later """
+
+    @classmethod
+    def iter_key_names(cls) -> Generator[str, None, None]:
```
[the remark this answers is not in the document]
Comment (happz) [reply tmt/utils.py]: Fixed in [cfb5922](https://github.com/teemtee/tmt/pull/1246/commits/cfb5922b66f9bddd3a131730e339b13ce7938f56).

**Review by psss — APPROVED** [review]
Thanks for the name improvement. Looks good. I've just noticed that `self.items()` could be directly used under `inject_to()` so I changed that plus wrapped long lines in 4c0d277.

**Pull request merged**